internal/instrumented_resp_connection/instrumented_resp_connection.go

```diff
@@ -20,48 +27,69 @@ func defaultCallbacks(logger *logger.Logger, logPrefix string) resp_connection.R
 			} else {
-				logger.Infof("%s%s %s", logPrefix, commandPrefix, command)
+				logger.Infof("%s %s", commandPrefix, command)
 			}
 		},
 		BeforeSendValue: func(value resp_value.Value) {
-			logger.Infof("%sSent %s", logPrefix, value.FormattedString())
+			logger.Infof("Sent %s", value.FormattedString())
 		},
 		BeforeSendBytes: func(bytes []byte) {
-			logger.Debugf("%sSent bytes: %q", logPrefix, string(bytes))
+			logger.Debugf("Sent bytes: %q", string(bytes))
 		},
 		AfterBytesReceived: func(bytes []byte) {
-			logger.Debugf("%sReceived bytes: %q", logPrefix, string(bytes))
+			logger.Debugf("Received bytes: %q", string(bytes))
 		},
 		AfterReadValue: func(value resp_value.Value) {
 			valueTypeLowerCase := strings.ReplaceAll(strings.ToLower(value.Type), "_", " ")
 			if valueTypeLowerCase == "nil" {
 				valueTypeLowerCase = "null bulk string"
 			}
-			logger.Debugf("%sReceived RESP %s: %s", logPrefix, valueTypeLowerCase, value.FormattedString())
-
+			logger.Debugf("Received RESP %s: %s", valueTypeLowerCase, value.FormattedString())
 		},
 	}
 }
 
-func NewFromAddr(logger *logger.Logger, addr string, connIdentifier string) (*resp_connection.RespConnection, error) {
-	logPrefix := ""
-	if connIdentifier != "" {
-		logPrefix = connIdentifier + ": "
+func NewFromAddr(logger *logger.Logger, addr string, connIdentifier string) (*InstrumentedRespConnection, error) {
+	newLogger := logger.Clone()
+	newLogger.PushSecondaryPrefix(connIdentifier)
+	c, err := resp_connection.NewRespConnectionFromAddr(addr, defaultCallbacks(newLogger))
+	if err != nil {
+		return nil, err
 	}
-	return resp_connection.NewRespConnectionFromAddr(
-		addr, connIdentifier, defaultCallbacks(logger, logPrefix),
-	)
+	return &InstrumentedRespConnection{
+		RespConnection: c,
+		logger:         newLogger,
+	}, nil
 }
 
-func NewFromConn(logger *logger.Logger, conn net.Conn, clientIdentifier string) (*resp_connection.RespConnection, error) {
-	logPrefix := ""
-	if clientIdentifier != "" {
-		logPrefix = clientIdentifier + ": "
+func NewFromConn(logger *logger.Logger, conn net.Conn, connIdentifier string) (*InstrumentedRespConnection, error) {
+	newLogger := logger.Clone()
+	newLogger.PushSecondaryPrefix(connIdentifier)
+	c, err := resp_connection.NewRespConnectionFromConn(conn, defaultCallbacks(newLogger))
+	if err != nil {
+		return nil, err
 	}
+	return &InstrumentedRespConnection{
+		RespConnection: c,
+		logger:         newLogger,
+	}, nil
+}
+
+func (c *InstrumentedRespConnection) GetIdentifier() string {
+	return c.logger.GetLastSecondaryPrefix()
+}
+
+// GetLogger returns a new logger with added secondary prefix: connection's identifier
+func (c *InstrumentedRespConnection) GetLogger() *logger.Logger {
+	return c.logger
+}
 
-	return resp_connection.NewRespConnectionFromConn(
-		conn, defaultCallbacks(logger, logPrefix),
-	)
+// UpdateLogger updates the connection's logger
+func (c *InstrumentedRespConnection) UpdateLogger(l *logger.Logger) {
```

Comment:
```suggestion
func (c *InstrumentedRespConnection) UpdateBaseLogger(l *logger.Logger) {
```

It could be confusing how we pass in a logger and a secondary prefix is automatically added to it. Looking at `GetLogger()` and `UpdateLogger(logger)`, I'd expect both to be symmetrical, i.e. after `UpdateLogger(logger)` is called with a logger, I'd get the same back in `GetLogger()`. Let's use the term `baseLogger` to make this distinction